include/field3d.hxx

```diff
@@ -167,7 +168,10 @@ class Field3D : public Field, public FieldData {
    * Note: the global "mesh" can't be passed here because
    * fields may be created before the mesh is.
    */
-  Field3D(Mesh *localmesh = nullptr);
+  Field3D(Mesh *localmesh = nullptr, CELL_LOC location_in=CELL_CENTRE,
+          DIRECTION xDirectionType_in=DIRECTION::Null,
+          DIRECTION yDirectionType_in=DIRECTION::Null,
+          DIRECTION zDirectionType_in=DIRECTION::Null);
```

Comment:
The reason is to ensure that the `fieldsCompatible` check would fail if setNullDirectionTypesToDefault() hasn't been called, which it should have been if the field has been allocated. Using `X`, `Y`, `Z` for defaults this wouldn't be true because `Y` is compatible with `Y`, `YOrthogonal` or `YAligned.